spec/requests/settings/user_deletes_account_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+describe 'Usuário deleta conta' do
+  it 'com sucesso' do
+    user = create(:user)
+    other_user = create(:user)
+    create(:post, user:)
+    user.personal_info.update(city: 'São Paulo')
+    user.education_infos.update(institution: 'Campus Code')
+    user.professional_infos.update(company: 'Rebase')
+    create(:invitation, profile: user.profile)
+    create(:invitation_request, profile: user.profile)
+    Connection.create(follower: user.profile, followed_profile: other_user.profile)
+    Connection.create(follower: other_user.profile, followed_profile: user.profile)
+
+    login_as user
+    delete(delete_account_path)
+
+    expect(User.all.count).to eq 1
+    expect(Profile.all.count).to eq 1
+    expect(Post.all.count).to eq 0
+    expect(Connection.all.count).to eq 0
+    expect(PersonalInfo.all.count).to eq 1
+    expect(EducationInfo.all.count).to eq 0
+    expect(ProfessionalInfo.all.count).to eq 0
+    expect(Invitation.all.count).to eq 0
+    expect(InvitationRequest.all.count).to eq 0
+    expect(flash[:notice]).to eq 'Conta excluída com sucesso'
+  end
+end
```

Comment:
Vcs acham que seria necessário fazer um teste de sistema onde eu não vejo mais o perfil deletado, acessando de outra conta?

Reply:
Após excluir a conta o perfil não existe mais no banco de dados.  Acho que seria desnecessário um teste de sistema para acessar um perfil que não existe, pois o rails já trata esses erros (record not found). Seria tipo tentar acessar uma rota (http://localhost:4000/profiles/9321931239)